tests/yadage/workspace-files.feature

```diff
@@ -0,0 +1,18 @@
+# Tests for the presence of the expected workflow files
+
+Feature: Workspace files
```

Comment:
This PR currently has:

```
tests
├── cwl
│   ├── log-messages.feature
│   └── workspace-files.feature
├── log-messages.feature
├── run-duration.feature
├── serial
│   ├── log-messages.feature
│   └── workspace-files.feature
├── snakemake
│   ├── log-messages.feature
│   └── workspace-files.feature
├── workspace-files.feature
└── yadage
    ├── log-messages.feature
    └── workspace-files.feature
```

Moreover, the feature names are often identical ("Workspace files"):

```console
$ grep -r 'Feature:' tests | sort
tests/cwl/log-messages.feature:Feature: Log messages
tests/cwl/workspace-files.feature:Feature: Workspace files
tests/log-messages.feature:Feature: Log messages
tests/run-duration.feature:Feature: Run duration
tests/serial/log-messages.feature:Feature: Log messages
tests/serial/workspace-files.feature:Feature: Workspace files
tests/snakemake/log-messages.feature:Feature: Log messages
tests/snakemake/workspace-files.feature:Feature: Workspace files
tests/workspace-files.feature:Feature: Workspace files
tests/yadage/log-messages.feature:Feature: Log messages
tests/yadage/workspace-files.feature:Feature: Workspace files
```

It may therefore not be clear to users whether the engine-specific test files "override" top-level directory test files or whether they complement them.

Two possibilities:

1.  Keep separate files, but rename the files and the feature names slightly, for example `tests/cwl/more-workspace-files.feature` and name the feature "Additional CWL-specific workspace files", so that users can say that they should look in both places.
2. Remove top-level files and copy/paste them into engine-specific files, so that users have only one place to look.

Each approach has pros and cons... See also remark https://github.com/reanahub/reana-demo-helloworld/pull/64#discussion_r1307361949 that we could take into account at the same time.

Reply:
Just like the helloworld repository, I chose option 2 mainly for tidiness reasons.